modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue

```diff
@@ -185,18 +199,44 @@ export default {
         }
       }
       if (!componentOnly) {
-        if (!s.startsWith(this.prefix)) {
-          if (this.prefix.startsWith(s)) {
-            // If they delete the `-`, and the prefix is `recipe-`,
-            // we want to restore `recipe-`, not set it to `recipe-recipe`
-            s = this.prefix;
-          } else {
-            s = this.prefix + s;
-          }
-        }
+        s = this.setPrefix(s);
       }
+
       return s;
     },
+    setPrefix (slug) {
+      // Get a fresh clone of the slug.
+      let updated = slug.slice();
```

Comment:
Strings are immutable objects, so you don't ever need to worry about cloning them. All operations on strings return a new immutable string.

Reply:
Bumping this lil' guy.